src/hw_isolation_record/manager.cpp

```diff
@@ -482,5 +482,53 @@ sdbusplus::message::object_path Manager::createWithEntityPath(
     return *entryPath;
 }
 
+std::optional<std::tuple<entry::EntrySeverity, entry::EntryErrLogPath>>
+    Manager::getIsolatedHwRecordInfo(
+        const sdbusplus::message::object_path& hwInventoryPath)
+{
+    // Make sure whether the given hardware inventory is exists
+    // in the record list.
+    auto entryIt =
+        std::find_if(_isolatedHardwares.begin(), _isolatedHardwares.end(),
+                     [hwInventoryPath](const auto& ele) {
+                         bool isIsolated = false;
+                         for (const auto& assocEle : ele.second->associations())
+                         {
+                             if ((std::get<0>(assocEle) == "isolated_hw") &&
+                                 (std::get<2>(assocEle) == hwInventoryPath.str))
+                             {
+                                 isIsolated = true;
+                                 break;
+                             }
+                         }
+
+                         // Make sure whether the given hardware inventory is
+                         // not resolved because the same hardware might be
+                         // isolated and resolved many times.
+                         if (isIsolated)
+                         {
+                             return !ele.second->resolved();
+                         }
+                         return false;
+                     });
+
+    if (entryIt == _isolatedHardwares.end())
+    {
+        return std::nullopt;
+    }
+
+    entry::EntryErrLogPath errLogPath;
+    for (const auto& assocEle : entryIt->second->associations())
+    {
+        if (std::get<0>(assocEle) == "isolated_hw_errorlog")
+        {
+            errLogPath = std::get<2>(assocEle);
```

Comment:
what if errLogPath is not found ? we still return the tuple at line 530 or return nullopt?

Reply:
The error log is optional i.e we won't generate an error log for the manual guard.

So, if the error log is not found we will return a tuple with the error log as empty (LN 520) and severity (always will be there) so the caller can take appropriate action)